### PR TITLE
feat: add support for automatic 997 generation

### PR DESCRIPTION
**Pull request by rsioss**
- add `x12-tools` dependency
- add new `AckTransactionSet` model, scoped to only `997` acks
- add new `acknowledgmentConfig` property on `TransactionSet` data models (excluding `AckTransactionSet`)
- generate and deliver 997 for inbound interchanges when any transaction set config for a transaction set in the input has acknowledgments enabled via the `acknowledgmentConfig` property

Closes: #19

### CODE REVIEW

File: src/setup/bootstrap/createStashRecords.ts [hunk 1]
```diff
@@ -16,7 +16,18 @@ export const createSampleStashRecords = async ({
 }: CreateSampleStashRecordsInput) => {
   const stashClient = buildStashClient();
 
+  const sftpBucketName = requiredEnvVar("SFTP_BUCKET_NAME");
+  const outboundBucketPath = "trading_partners/ANOTHERMERCH/outbound";
+
   const partnership: Partnership = {
+    ack: {
+      enabled: true,
+      destination: {
```
Comment (BDQ) [inline src/setup/bootstrap/createStashRecords.ts:25]: Did you consider putting a `997` entry in the `transactionSets` array? That would keep any "destinations" config in the same place, and allow multiple for 997.

**Review by BDQ — APPROVED** [review]
Looks great, left one question regarding 997 destinations config

File: src/setup/bootstrap/createStashRecords.ts (hunk 1 above)
Comment (rsioss) [reply src/setup/bootstrap/createStashRecords.ts:25]: I'm not sure how that would work, actually. The `997`s don't require the use of a guide, so resolving the config / identifying the transaction set wouldn't work with out current mechanisms. What were you envisioning it to look like as an entry in `transactionSets`?

Also not sure what you meant by multiple 997 -- did you mean different content for different partnerships? (again with no guide and the use of `x12-tools` to generate the `997` content this doesn't seem necessary IMO)

Comment (BDQ) [reply src/setup/bootstrap/createStashRecords.ts:25]: I didn't have a design in mind, but maybe "guideId" would become optional - we need that to allow base guides anyway?

By "allow multiple for 997" - I meant destinations - like RT today send 997s to the buckets (the default you've covered) but also send 997 to their data warehouse (stitch).

Comment (rsioss) [reply src/setup/bootstrap/createStashRecords.ts:25]: Ok, refactored to align with the partner profiles data model (and to handle 997 acks as another type of transaction set). PTAL when you have a chance, @BDQ ! Thanks again for the feedback and for kicking ideas around on how this should look

**Review by BDQ — CHANGES_REQUESTED** [review]
I tried to take this for a test drive in my account, and hit an error running `npm run bootstrap`
image

**Comment by rsioss** [comment]
> I tried to take this for a test drive in my account, and hit an error running `npm run bootstrap` 

🤦 apologies, @BDQ! I had already addressed this actually, but apparently I never pushed [the commit](https://github.com/Stedi-Demos/bootstrap/pull/38/commits/2d05ed50e6805b1bedf9992d84a3bbaa633af6fd) up! Give it another shot when you have time

**Review by BDQ — APPROVED** [review]

**Pull request merged**